src/main/java/com/powsybl/openloadflow/network/action/LfGeneratorAction.java

```
@@ -0,0 +1,50 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.network.action;
+
+import com.powsybl.action.GeneratorAction;
+import com.powsybl.openloadflow.network.*;
+import com.powsybl.openloadflow.util.PerUnit;
+
+import java.util.Optional;
+import java.util.OptionalDouble;
+
+/**
+ * @author Bertrand Rix {@literal <bertrand.rix at artelys.com>}
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ * @author Jean-Luc Bouchot {@literal <jlbouchot at gmail.com>}
+ */
+public class LfGeneratorAction extends AbstractLfAction<GeneratorAction> {
+
+    public LfGeneratorAction(String id, GeneratorAction action) {
+        super(id, action);
+    }
+
+    @Override
+    public boolean apply(LfNetwork network, LfContingency contingency, LfNetworkParameters networkParameters) {
+        LfGenerator generator = network.getGeneratorById(action.getGeneratorId());
+        if (generator != null) {
+            OptionalDouble activePowerValue = action.getActivePowerValue();
+            Optional<Boolean> relativeValue = action.isActivePowerRelativeValue();
+            if (relativeValue.isPresent() && activePowerValue.isPresent()) {
+                if (!generator.isDisabled()) {
+                    double change = activePowerValue.getAsDouble() / PerUnit.SB;
+                    double newTargetP = Boolean.TRUE.equals(relativeValue.get()) ? generator.getTargetP() + change : change;
+                    generator.setTargetP(newTargetP);
+                    generator.setInitialTargetP(newTargetP);
+                    generator.reApplyActivePowerControlChecks(networkParameters, null);
+                    return true;
+                }
+            } else {
+                throw new UnsupportedOperationException("Generator action on " + action.getGeneratorId() + " : configuration not supported yet.");
+            }
```

By checking this at apply time and not at creation time, you loose the fail fast property that existed before.

Before:
   If there was an issue it was raised at launch time. User could see it immediately.
Witht his PR:
   FOr a long computation, the issue may happen hours after launch time/
    => User looses time
    => The user may not see the issue if lost in a large log.

---

Indeed this can probably be moved at construction time. I will do it.